src/cljs_bean/core.cljs

```diff
@@ -200,12 +225,6 @@
   (-invoke [coll k not-found]
     (-lookup coll k not-found))
 
-  IEditableCollection
-  (-as-transient [_]
-    (let [result (volatile! (transient {}))]
-      (gobj/forEach obj (fn [v k _] (vswap! result assoc! (prop->key k) v)))
-      @result))
```

Comment:
Perhaps a transient variant of `Bean` could be created by this that essentially allows you to mutate its held JavaScript object without cloning.